[18.0][FIX] product_sequence: test disable

eduezerouali-tecnativa opened the pull request:
Since odoo 16.0 the template is copied instead and its first variant is returned.
https://github.com/OCA/OCB/blob/9fe44f03ba29e98e4bbc103983125e1393e45358/addons/product/models/product_product.py#L460

So that part should be move to `product.template`  and that is the reason why test were failing in the repository.

This should be backported?

Please could you review @juancarlosonate-tecnativa @pedrobaeza @victoralmau

pedrobaeza reviewed (commented):
`copy_data` should be implemented at `product.product` level, not `product.template`.

juancarlosonate-tecnativa commented:
I think the issue is not related to the copy, the problem lies in the create. If you look at the test error, when the product is created with the default_code, it’s set as PROD02 in the test, but during the copy, the sequence uses /, completely ignoring the product’s original default_code. Because of that, it treats / as the default value and generates the next sequence, which causes the copied product not to match the expected default_code or the copy logic. I believe the default_code from the template should be taken and assigned as the value to make it work correctly.

eduezerouali-tecnativa commented:
@pedrobaeza are you 100% sure to change `copy` to `copy_data` ? https://github.com/odoo/odoo/pull/154132/files#diff-35eeb9b2264b76eca30fcb0a923757dde3ca14c636ce6faf66ef31a6b649799bL443

I do agree with @juancarlosonate-tecnativa. I found that the problem is when it tries to generate the first variant when coping the template.

pedrobaeza commented:
OK, I guessed that the problem was that the method was not intervening on the proper place, but the aim is to fix the tests, so go ahead with any proposal to fix them.

pedrobaeza commented:
@eduezerouali-tecnativa please finish this to get green CI. If not able to fix it, just disable current tests.

eduezerouali-tecnativa committed:
[FIX] product_sequence: disable test

Disable 2 test that were making branch to fail, not easy solution has been found, so diasable for now.

eduezerouali-tecnativa commented:
ping @pedrobaeza

pedrobaeza commented on product_sequence/models/product_product.py:
```diff
@@ -38,6 +38,7 @@ def create(self, vals_list):
             else:
                 vals_list_updated.append(vals)
         res = super().create(vals_list_updated)
+
```
Remove this extra line.

pedrobaeza reviewed (commented):
Indicate in the extended description of the commit message that they are making the branch to fail, and no easy solution has been found, so they are disables for now.

pedrobaeza reviewed (approved):
/ocabot merge patch
